drivers/hmis/app/models/hmis/ce/opportunity.rb

```diff
@@ -17,10 +17,10 @@ class Opportunity < GrdaWarehouseBase
                primary_key: 'identifier',
                class_name: 'Hmis::WorkflowDefinition::Template'
 
-    has_many :referrals, class_name: 'Hmis::Ce::Referral', dependent: :restrict_with_exception
+    has_many :referrals, class_name: 'Hmis::Ce::Referral' # use default dependent behavior, since opportunities are soft-deleted
```

Comment:
I think this comment is a little confusing. For my own understanding, the default behavior is: when an Opportunity is destroyed, the associated referrals continue to reference the soft-deleted opportunity record (in `ce_referrals` `opportunity_id` column)

Maybe this is clearer?

```suggestion
    has_many :referrals, class_name: 'Hmis::Ce::Referral' # Referrals continue to reference the opportunity even after it is soft-deleted, as no dependent behavior is specified.
```